### PR TITLE
[SYCL] Use SPIR calling convention when inserting itt calls

### PR DESCRIPTION
**Pull request by romanovvlad**
The definitions of ITT functions has spir_func calling convention,
so calling them using C calling convention is not correct.

### CODE REVIEW

File: llvm/lib/Transforms/Instrumentation/SPIRITTAnnotations.cpp [hunk 1]
```diff
@@ -153,7 +153,12 @@ Instruction *emitCall(Module &M, Type *RetTy, StringRef FunctionName,
   auto *FT = FunctionType::get(RetTy, ArgTys, false /*isVarArg*/);
   FunctionCallee FC = M.getOrInsertFunction(FunctionName, FT);
   assert(FC.getCallee() && "Instruction creation failed");
-  return CallInst::Create(FT, FC.getCallee(), Args, "", InsertBefore);
+  Function *F = dyn_cast<Function>(FC.getCallee());
+  assert(F);
```
Comment (AlexeySachkov) [inline llvm/lib/Transforms/Instrumentation/SPIRITTAnnotations.cpp:157]:
```suggestion
  Function *F = cast<Function>(FC.getCallee());
```

**Review by AlexeySachkov — APPROVED** [review]
Would be good to have a brief description of a problem in the PR description

File: llvm/lib/Transforms/Instrumentation/SPIRITTAnnotations.cpp [hunk 2]
```diff
@@ -153,7 +153,13 @@ Instruction *emitCall(Module &M, Type *RetTy, StringRef FunctionName,
...
+  F->setCallingConv(CallingConv::SPIR_FUNC);
```
Comment (bader) [inline llvm/lib/Transforms/Instrumentation/SPIRITTAnnotations.cpp:158]: This is valid only for modules with spirv* target triple. I suggest we test this assumption with the assert at the beginning of the pass.

Comment (romanovvlad) [reply llvm/lib/Transforms/Instrumentation/SPIRITTAnnotations.cpp:157]: It seems all passes in llvm/lib/SYCLLowerIR assume spir target triple, but they do not have such a check. Suggest that it's done for all passes separately.

Comment (bader) [reply llvm/lib/Transforms/Instrumentation/SPIRITTAnnotations.cpp:157]: > It seems all passes in llvm/lib/SYCLLowerIR assume spir target triple, but they do not have such a check. 

First, SPIRITTAnnotations.cpp is not located in llvm/lib/SYCLLowerIR, so I would not mix them together.
Second, not all SYCLLowerIR passes do not assume spir target triple. E.g. https://github.com/intel/llvm/blob/sycl/llvm/lib/SYCLLowerIR/TargetHelpers.cpp. 
According to the readme:

```
A collection of SYCL language-specific LLVM IR transformation passes applied to
the result of clang code generation. They bring the IR to a valid state from
the SYCL semantics perspective.
```

It sounds like this is the place to keep passes for non-SPIR-V targets as well.

> Suggest that it's done for all passes separately.

Let's add check for SPIRITTAnnotations.cpp in this PR and for llvm/lib/SYCLLowerIR passes separately.

**Review by elizabethandrews — APPROVED** [review]

**Review by bader — APPROVED** [review]

**Pull request merged**